### DISCUSSION TITLE
Jetson Nano with Raspberry Pi Camera

### DISCUSSION THREAD

**Discussion by ucaglarcaliskan** [svpcom/wfb-ng#161] [Questions] [answered]
Hello everyone,

We try to use Jetson Nano with Raspberry Pi V2 camera as drone and an ubuntu as gs. When we try to run Logitech c920 command, we get an error like 

> Device is not a valid uvc h264 camera.

 What gstreamer command should we use for raspberry pi camera on Jetson for drone and gs? Could you help us for that?

**Comment by svpcom** [comment]
GS pipeline for C920:
```
gst-launch-1.0 uvch264src device=/dev/video0 initial-bitrate=4000000 average-bitrate=4000000 iframe-period=3000 \
              name=src auto-start=true src.vidsrc ! queue ! video/x-h264,width=1280,height=720,framerate=30/1 \
            ! h264parse ! rtph264pay ! udpsink host=localhost port=5602
```

where **/dev/video0 is C920-related device**. Check it with `v4l2-ctl -D -d /dev/video0`

**Reply by ucaglarcaliskan** [reply]
We tried this pipeline but we get "Device is not a valid uvc h264 camera.". We use raspberry pi V2 camera and this is v4l2-ctl output:

> Driver Info (not using libv4l2):
	Driver name   : tegra-video
	Card type     : vi-output, imx219 6-0010
	Bus info      : platform:54080000.vi:0
	Driver version: 4.9.253
	Capabilities  : 0x84200001
		Video Capture
		Streaming
		Extended Pix Format
		Device Capabilities
	Device Caps   : 0x04200001
		Video Capture
		Streaming
		Extended Pix Format

**Reply by svpcom** [reply]
Try other /dev/video* devices. C920 may not be the first device.
For RPI camera try to use something like:
```
gst-launch-1.0 -v v4l2src device=/dev/video0 ! 'video/x-raw, width=(int)1280, height=(int)720, format=(string)I420' ! omxh264enc bitrate=4000000 !  rtph264pay config-interval=1 ! udpsink host=localhost port=5602
```

**Reply by ucaglarcaliskan** [reply]
Thank you for your answer. We tried this now and get this error: 

> Setting pipeline to PAUSED ...
Pipeline is live and does not need PREROLL ...
ERROR: from element /GstPipeline:pipeline0/GstV4l2Src:v4l2src0: Internal data stream error.
Additional debug info:
gstbasesrc.c(3055): gst_base_src_loop (): /GstPipeline:pipeline0/GstV4l2Src:v4l2src0:
streaming stopped, reason not-negotiated (-4)
ERROR: pipeline doesn't want to preroll.
Setting pipeline to PAUSED ...
Setting pipeline to READY ...
Setting pipeline to NULL ...
Freeing pipeline ...

**Comment by ucaglarcaliskan** [comment, marked as answer]
Hi, It was a little late but I wanted to write the solution here. We use this (it is multiudpsink but same logic):

`gst-launch-1.0 -e nvarguscamerasrc ! 'video/x-raw(memory:NVMM), width=640, height=480, framerate=25/1' ! nvv4l2h264enc bitrate=400000 insert-sps-pps=true ! rtph264pay mtu=1400 ! multiudpsink clients=127.0.0.1:5602,127.0.0.1:5603 sync=false async=false`
